### ISSUE TITLE
LDAP groups are being unchecked when navigating back and forth to the admin page

### ISSUE THREAD

**Issue by jnfrmarks** [owncloud/core#13904]
### Steps to reproduce

1. Install owncloud 
2. Enable LDAP
3. Go to admin page and configure the LDAP server to find users from multiple groups
4. navigate to users page
5. See all of the users and group names though no group numbers nor are the users in a group
6. Navigate back to the admin page and tab through the ldap configuration
7. Only one group is checked even though in step 3 multiple groups were selected
8. Go back to the users page
9. Group now shows up with a user count
10. Group that was not checked in the ldap configuration shows up on the users page

### Expected behaviour

1.  Group user count should show up
2.  Group check boxes should not change between navigation to the admin page
3.  Groups on user page should match groups checked in the ldap configuration

### Actual behaviour

1.  Group user count does not show up on the first viewing of the users page
2.  Group check boxes change between navigation to the admin page
3.  Groups on user page do not match groups checked in the ldap configuration (this happens after the error in step 2 happens and is most likely a side affect of what is causing that error)

step 2:

![p280](https://cloud.githubusercontent.com/assets/8480343/6046437/34895ec2-ac70-11e4-8848-709e014ac4ba.PNG)

step 3:

![p281](https://cloud.githubusercontent.com/assets/8480343/6046439/3a317062-ac70-11e4-9de7-4b39757c089f.PNG)

step 3:

![p282](https://cloud.githubusercontent.com/assets/8480343/6046441/3f011250-ac70-11e4-95b2-6ccbdccfd5f0.PNG)

step 5:

![p283](https://cloud.githubusercontent.com/assets/8480343/6046446/42e0a4f8-ac70-11e4-81ca-ac343a91a77d.PNG)

step 6:

![p284](https://cloud.githubusercontent.com/assets/8480343/6046453/4e283b32-ac70-11e4-9321-34160338e776.PNG)

step 6:

![p285](https://cloud.githubusercontent.com/assets/8480343/6046461/590276f8-ac70-11e4-8ece-d2591d98d4cd.PNG)

step 9:

![p286](https://cloud.githubusercontent.com/assets/8480343/6046464/60e3be90-ac70-11e4-8cd7-a27ae8fac718.PNG)

### Server Host info:

Ubuntu 14.04
PHP 5.5.9

### Server configuration

ownCloud Enterprise Edition 8.0 RC 1 (daily) Build:2015-02-04T04:19:41+00:00 23c6a0cf99818c986f5a6537cf765fa9fdff5757)
plus patch for 13740 and 13897.  

I don't believe that the patches are the cause of what I'm seeing

**Comment by jnfrmarks** [comment]
I only see this the first time I configure the LDAP.  If I go back to the configuration and add back in the missing groups, then everything is as expected on the users page and the groups remain checked when navigating back and forth between the admin and users page.

**Comment by DeepDiver1975** [comment]
@blizz please have a look and rate the criticality - THX

scheduling for 8.0 to get this analysed asap

**Comment by karlitschek** [comment]
@blizzz

**Comment by blizzz** [comment]
I was able to reproduce it after a screenshare with @jnfrmarks and subsequently able to find out where the problem origins.

The basic problem is that the group-member-association attribute is first detected correctly, but not adjusted in the GUI. There it is placed in the advanced tab, and upon the change there it will be overridden with a wrong value. Therefore, no group memberships are visible and groups appear empty. Also, the save action causes that one group is lost in the user and group tab. Upon adding them again, the group membership will also be detected and fixed, so that the effect does not show up anymore. 

Long story short:
* [ ] group-member-association attribute is not set in the GUI after detection
* [ ] save on Adanved (probably also Expert) tab leads to loss of one selected group in each user and group tab.

Need to check whether the same happens on OC 7.

**Comment by blizzz** [comment]
OC 7 is also affected. Therefore, not a new bug and thus not critical for OC 8, move milestone → to 8.0.1. Agreed?

**Comment by DeepDiver1975** [comment]
> to 8.0.1. Agreed?

agreed - THX

**Comment by blizzz** [comment]
8.1 should not suffer this, as a fix is included in https://github.com/owncloud/core/pull/15492, but needs to be done for older versions.

**Comment by blizzz** [comment]
@jnfrmarks this should be fixed in master for 8.1. Can you confirm?

**Comment by jnfrmarks** [comment]
retest blocked by #15866

**Comment by blizzz** [comment]
Suggested fix for #15866 is in https://github.com/owncloud/core/pull/16122

**Comment by blizzz** [comment]
… and the blocking issue is fixed in master :cake:

**Comment by cmonteroluque** [comment]
Fixed in 8.1
